src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -208,6 +205,66 @@ public static HostString FromUriComponent(Uri uri)
                 UriComponents.HostAndPort, UriFormat.Unescaped));
         }
 
+        /// <summary>
+        /// Matches the host portion of a host header value against a list of patterns.
+        /// The host may be the encoded punycode or decoded unicode form so long as the pattern
+        /// uses the same format.
+        /// </summary>
+        /// <param name="value">Host header value with or without a port.</param>
+        /// <param name="patterns">A set of pattern to match, without ports.</param>
+        /// <remarks>
+        /// The port on the given value is ignored. The patterns should not have ports.
+        /// The patterns may be exact matches like "example.com", a top level wildcard "*"
+        /// that matches all hosts, or a subdomain wildcard like "*.example.com" that matches
+        /// "abc.example.com:443" but not "example.com:443".
+        /// Matching is case insensitive.
+        /// </remarks>
+        /// <returns></returns>
+        public static bool MatchesAny(string value, IList<string> patterns)
```

Comment:
Any reason these aren't StringSegments?

Reply:
Neither are expected to be pre processed, only raw input.

Reply:
This should be a StringSegment. This is a generic helper API that should be able to work on slices of strings. 

Ideally it would be `ReadOnlySpan<char>` but we haven't taken on that dependency widely as yet.